starboard/common/device_type.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Module Overview: Starboard device type module
+//
+// Defines Type of the device, e.g. such as "TV", "STB", "OTT".
+// Please see Youtube Technical requirements for a full list
+// of allowed values.
+
+#ifndef STARBOARD_COMMON_DEVICE_TYPE_H_
+#define STARBOARD_COMMON_DEVICE_TYPE_H_
+
+namespace starboard {
+
+// Blue-ray Disc Player (BDP).
+const char kSystemDeviceTypeBlueRayDiskPlayer[] = "BDP";
```

Comment:
Let's make these literals defined and initialized in the device_type.cc file and just use extern declaration in the header. Otherwise we'll end up with multiple symbols defined as every object file that includes the header would define a symbol.

Heres is a good example:
https://github.com/youtube/cobalt/blob/main/cobalt/browser/switches.h#L27
https://github.com/youtube/cobalt/blob/main/cobalt/browser/switches.cc#L27

Reply:
Done

Reply:
Really nice refactor / cleanup on this. Everything is in single file with all the comments right there as well.